### PR TITLE
Added pacaur support to the arch linux plugin.

### PR DESCRIPTION
**Pull request by Majora320**
I added pacaur compatibility to the arch linux plugin, and refactored a bit of the logic.

### CODE REVIEW

**Comment by slavaGanzin** [comment]
@Majora320 Maybe you will be interesed to add this into plugin:

https://wiki.archlinux.org/index.php/Reflector

```sudo reflector --verbose -l 10 --sort rate --save /etc/pacman.d/mirrorlist &```

**Comment by Majora320** [comment]
Reflector has so many useful options (country code, etc.) that I think a custom alias created by the user would be more useful than a prescripted plugin @slavaGanzin, but good idea all the same. I should look into some more arch linux tools and perhaps add them here.

File: plugins/archlinux/archlinux.plugin.zsh [hunk 1]
```diff
@@ -35,6 +27,47 @@ else
...
+  upgrade() {
+    pacaur -Syu
+  }
+elif (( $+commands[yaourt] )); then
+  upgrade() {
+    yaourt -Syu
+  }
+else
+  upgrade() {
+    pacman -Syu
```
Comment (mcornella) [inline plugins/archlinux/archlinux.plugin.zsh]: You left the `sudo` here.

**Comment by mcornella** [comment]
This is a duplicate of #4263, which fell out of my mind. That one has become corrupted after the big archlinux refactor, but I'd like to check if those are similar aliases and to add them to the README, as that other one does.

**Comment by johnramsden** [comment]
I'd appreciate this being added. ```pacaur``` is far better than ```yaourt``` and should be considered a first class citizen.

**Comment by mcornella** [comment]
Yes, as I said, I'm waiting for the README modification.

**Comment by Majora320** [comment]
I'm editing to add the commands to the README and the `sudo` to the last else branch now.

**Comment by mcornella** [comment]
💯 no need to hurry!

**Comment by Majora320** [comment]
Sorry, I sounded kind of offensive, didn't mean to be hostile :)

**Comment by Majora320** [comment]
Alright, done.

**Comment by Majora320** [comment]
Wait never mind, I left `### pacaur` lowercase :P

**Comment by Majora320** [comment]
Alright, now I can say that I'm done.

File: plugins/archlinux/archlinux.plugin.zsh (hunk 1 above)
Comment (Majora320) [inline plugins/archlinux/archlinux.plugin.zsh]: Correct, fixed.

**Comment by johnramsden** [comment]
Thanks!

**Comment by mcornella** [comment]
Nice job thanks!

**Pull request closed without merging**